### PR TITLE
allow to print solver's log in file

### PR DESCRIPTION
**Pull request by a-zakir**

### CODE REVIEW

File: src/python/antares_xpansion/benders_driver.py [hunk 1]
```diff
@@ -63,6 +70,27 @@ def set_simulation_output_path(self, simulation_output_path: Path):
...
             )
 
+    def _run_and_print_in_std_out_err_and_file(self):
+        with open(self._benders_log_file, "w") as benders_log:
+            with subprocess.Popen(
+                self._get_solver_cmd(), shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+            encoding='utf-8') as proc:
+                for output in proc.stdout:
+                    benders_log.write(output)
+
```
Comment (sgatto) [inline src/python/antares_xpansion/benders_driver.py]: there should also  be

`sys.stdout.write(output)`

**Review by sgatto — APPROVED** [review]

File: tests/python/test_benders_driver.py [hunk 2]
```diff
@@ -208,6 +209,108 @@ def test_clean_benders_step_if_not_keep_mps(self, tmp_path):
...
+        my_install_dir = Path("Dummy/Path/to/")
+        my_n_mpi = 13
+        exe_path = os.path.normpath(os.path.join(my_install_dir, my_benders_mpi))
+
+        benders_driver = BendersDriver(exe_path, "", "")
+
+        simulation_output_path = tmp_path
+        lp_path = Path(os.path.normpath(os.path.join(simulation_output_path, 'lp')))
+        lp_path.mkdir()
+        os.chdir(lp_path)
```
Comment (tbittar) [inline tests/python/test_benders_driver.py]: Could we refactor this part of code which is the same in the three tests (in a setup method for example) ?

File: tests/python/test_benders_driver.py [hunk 3]
```diff
@@ -208,6 +209,108 @@ def test_clean_benders_step_if_not_keep_mps(self, tmp_path):
...
+                
+
+
+        assert benders_log_file.exists()
+        assert benders_log_file.stat().st_size == len(expected_std_out[0])
+        with open(benders_log_file) as log:
+            lines = [line for line in log.readlines()]
+            assert lines == expected_std_out  
+
+    def test_benders_that_stderr_are_printed_in_log_file(self, tmp_path):
```
Comment (tbittar) [inline tests/python/test_benders_driver.py]: This test is identical to the previous except that it involves `stderr` in place of `stdout`, is it possible to write a unique test that "takes the stream as parameter" ?

File: src/python/antares_xpansion/config_loader.py [hunk 4]
```diff
@@ -356,6 +356,10 @@ def log_level(self):
             "log_level", self._config.settings_default["log_level"]
         )
         return int(log_level_str) 
+    
+    def benders_log_file(self)-> Path:
```
Comment (tbittar) [inline src/python/antares_xpansion/config_loader.py:360]: This function has the same name as `benders_log_file` in `benders_driver.py` line 170, I don't know if this is annoying

**Pull request merged**